src/RageInputDevice.h

```diff
@@ -315,11 +315,14 @@ enum DeviceButton
 RString DeviceButtonToString( DeviceButton i );
 DeviceButton StringToDeviceButton( const RString& s );
 
+int device_id_handler(const std::string& itg_device_persistent_id);
+
 struct DeviceInput
 {
 public:
 	InputDevice device;
 	DeviceButton button;
+	int uniqueID;
```

Comment:
what is this supposed to mean in the context of the input event? Each event is already uniquely identified by the tuple (device, button,  bDown, ts)

Reply:
"device" is specific enough until a controller disconnects, then when it is plugged in again it's treated as a different controller. This aims to prevent that by associating an input device with a unique ID.